test/e2e/framework/helper/certificatesigningrequests.go

```diff
@@ -40,6 +41,11 @@ func (h *Helper) WaitForCertificateSigningRequestSigned(name string, timeout tim
 			if err != nil {
 				return false, fmt.Errorf("error getting CertificateSigningRequest %s: %v", name, err)
 			}
+
+			if util.CertificateSigningRequestIsFailed(csr) {
+				return false, fmt.Errorf("CertificateSigningRequest is failed: %v", csr.Status)
```

Comment:
nit: "is failed" does not read well
```suggestion
				return false, fmt.Errorf("CertificateSigningRequest has failed: %v", csr.Status)
```
or
```suggestion
				return false, fmt.Errorf("CertificateSigningRequest failed: %v", csr.Status)
```